### PR TITLE
Fix denial of service with large numbers in cpuset-cpus and cpuset-mems

### PR DESCRIPTION
**Pull request by thaJeztah**
Using a value such as `--cpuset-mems=1-9223372036854775807` would cause
`dockerd` to run out of memory allocating a map of the values in the
validation code. Set limits to the normal limit of the number of CPUs,
and improve the error handling.

Reported by Huawei PSIRT.

**- Description for the changelog**

```Markdown
* Fix denial of service with large numbers in `--cpuset-cpus` and `--cpuset-mems`
```

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @justincormack @AkihiroSuda @vdemeester PTAL 🤗

File: pkg/parsers/parsers.go [hunk 1]
```diff
@@ -31,7 +32,7 @@ func ParseKeyValueOpt(opt string) (string, string, error) {
 //     03,1-3      <- this is gonna get parsed as [1,2,3]
 //     3,2,1
 //     0-2,3,1
-func ParseUintList(val string) (map[int]bool, error) {
+func ParseUintList(val string, maximum int) (map[int]bool, error) {
```
Comment (vdemeester) [inline pkg/parsers/parsers.go]: Should we keep the signature (for not breaking downstream-dependencies) and add a new method `ParsUintListMaximum` or something ?

Comment (thaJeztah) [reply pkg/parsers/parsers.go]: Ah, yes; I could add a new one.

**Comment by thaJeztah** [comment]
@vdemeester pushed an extra commit; if it looks good, I'll squash, and update my backports 😅

**Review by vdemeester — APPROVED** [review]
LGTM :tiger:

**Comment by thaJeztah** [comment]
Squashed; this should be ready to go

ping @justincormack PTAL

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**